Tests/CommandsTests/TestToolTests.swift

```diff
@@ -30,6 +31,37 @@ final class TestToolTests: XCTestCase {
         XCTAssert(try execute(["--version"]).contains("Swift Package Manager"))
     }
 
+    func testAllTestCases() throws {
+        #if os(macOS)
+        fixture(name: "Miscellaneous/ParallelTestsPkg") { path in
+            let result: String
+            do {
+                result = try execute([], packagePath: path)
+            } catch SwiftPMProductError.executionFailure(_, _, let stderr) {
+                result = stderr
+            }
+            XCTAssert(result.contains("Test Suite 'ParallelTestsFailureTests' failed"))
+            XCTAssert(result.contains("Test Suite 'ParallelTestsTests' passed"))
+            XCTAssert(result.contains("Executed 3 tests, with 1 failure"))
+        }
+        #endif
+    }
+
+    func testFilteredTestCase() throws {
```

Comment:
We probably already have such integration tests, can you update the existing ones as these are costly to run?

Reply:
Oh right, I didn't notice the integration tests. There is currently a test which invokes `swift test` (`XCBuildTests.testSwiftTest`) but it seems to be specific to the Xcode build system, so I've added a new one – `BasicTests.testSwiftTest`.